section/ontology.md

```diff
@@ -413,3 +413,238 @@ ex:leftPadMapping
     fno:returnMapping    [ a                   fnom:DefaultReturnMapping ;
                            fnom:functionOutput ex:outputStringOutput ] .
 ```
+
+### Function Composition {#composition}
+
+
+
+As an alternative to providing an <a>implementation</a> of a <a>function</a>, we can also define how it can be realized 
+using other <a>function</a>s, namely by defining a <a>composition</a>. To illustrate this point, we first define a 
+function, `ex:sum3Function`, that computes the sum of three integers. We are using the parameter and output definitions made earlier, 
+so we only have to provide the third parameter, `ex:intParameterC`. For brevity, we omit the corresponding <a>problem</a> and <a>algorithm</a>. 
+
+```turtle "example": " "
+ex:sum3Function
+    a                   fno:Function ;
+    fno:name            "Sum3"^^xsd:string ;
+    dcterms:description "This function calculates the sum of three integers."^^xsd:string ;
+    fno:expects ( ex:intParameterA ex:intParameterB ex:intParameterC ) ;
+    fno:returns ( ex:sumOutput ) 
+
+ex:intParameterC
+    a             fno:Parameter ;
+    fno:predicate ex:sumValue2 ;
+    fno:type      xsd:integer ;
+    fno:required  "true"^^xsd:boolean .
+```
+
+##### `fnoc:Composition` {#fnoc-composition}
+
+We will realize `ex:sum3Function` (defined in the previous paragraph) using `ex:sumFunction` introduced earlier, as illustrated with the following pseudo code: 
+```
+    ex:sum3Function(intParameterA, intParameterB, intParameterC) =  
+        ex:sumFunction(ex:sumFunction(intParameterA, intParameterB), intParameterC) 
+```
+
+In this construct, we use `ex:sumFunction` twice. In order to be able to reference the two distinct calls of the
+function `ex:sumFunction` unambiguously, we first create two distinct *applications* 
+of it using <a href="#fnoc-applies">`fnoc:applies`</a>. Then, we define the composition using a set of <a>composition mapping</a>s.  
```

Comment:
Is this application always needed? Or only in the case of using the same function multiple times?

Reply:
I think it is only needed if we want to use the same function multiple times in a composition. Will clarify in the text.

Reply:
done